package/gluon-mesh-batman-adv-core/files/lib/gluon/announce/statistics.d/clients

```diff
@@ -1,20 +1,51 @@
-local list = io.lines("/sys/kernel/debug/batman_adv/bat0/transtable_local")
+local iwinfo = require 'iwinfo'
+
+local counts = { total = 0
+               , wifi = 0
+               , wifi24 = 0
+               , wifi5 = 0
+               }
 
-local count = 0
-local wifi = 0
+local list = io.lines("/sys/kernel/debug/batman_adv/bat0/transtable_local")
 for line in list do
   local mac, _, flags, lastseen = line:match("^ %* ([0-9a-f:]+) *(.- )%[(.-)%] +(%d+%.%d+)")
   if mac then
     if not flags:match('P') then
-      count = count + 1
+      counts.total = counts.total + 1
 
       if flags:match('W') then
-        wifi = wifi +1
+        counts.wifi = counts.wifi +1
       end
     end
   end
 end
 
-return { total = count
-       , wifi  = wifi
-       }
+function count_iface_stations(iface)
+  local wifitype = iwinfo.type(iface)
+  if wifitype == nil then
+    return
+  end
+
+  local freq = iwinfo[wifitype].frequency(iface)
+  local key
+  if freq >= 2400 and freq < 2500 then
+    key = "wifi24"
+  elseif freq >= 5000 and freq < 6000 then
+    key = "wifi5"
+  else
+    return
+  end
+
+  for k, v in pairs(iwinfo[wifitype].assoclist(iface)) do
+    counts[key] = counts[key] + 1
+  end
```

Comment:
No, it can't. The length operator only really works on numerically indexed tables without gaps. From [the Lua manual](http://www.lua.org/manual/5.1/manual.html#2.5.5):
> The length of a table t is defined to be any integer index n such that t[n] is not nil and t[n+1] is nil; moreover, if t[1] is nil, n can be zero. For a regular array, with non-nil values from 1 to a given n, its length is exactly that n, the index of its last value. If the array has "holes" (that is, nil values between other non-nil values), then #t can be any of the indices that directly precedes a nil value (that is, it may consider any such nil value as the end of the array). 

The `assoclist` however is indexed by MAC addresses, thus `#` doesn't work.